File: benchmarks/language_compare/function_calls.c
/* BENCH_ITERATIONS=1000000 BENCH_EXPECTED=7270106 */
#include <stdint.h>
#include <stdio.h>
static int64_t gcd(int64_t a, int64_t b) { return b == 0 ? a : gcd(b, a % b); }
int main(void) {
    int64_t state = 1, checksum = 0;
    for (int64_t i = 0; i < 1000000; i++) {
        state = (state * 48271) % 2147483647;
        checksum = (checksum + gcd(state, state % 10000 + 12345)) % 1000000007;
    }
    printf("%lld\n", (long long)checksum);
    return 0;
}

File: benchmarks/language_compare/function_calls.rs
// BENCH_ITERATIONS=1000000 BENCH_EXPECTED=7270106
fn gcd(a: i64, b: i64) -> i64 { if b == 0 { a } else { gcd(b, a % b) } }
fn main() {
    let mut state = 1_i64;
    let mut checksum = 0_i64;
    for _ in 0..1_000_000 {
        state = (state * 48_271) % 2_147_483_647;
        checksum = (checksum + gcd(state, state % 10_000 + 12_345)) % 1_000_000_007;
    }
    println!("{checksum}");
}
